src/main/java/utility/exceptions/DuplicateUsernameException.java

```diff
@@ -0,0 +1,7 @@
+package utility.exceptions;
```

Comment:
This exception (along with the ones you defined below), are specific to the registration use case and should not be put under `utility`. You should define them as static classes within `RegisterInteractor` (see how we do that in `BuyStockInteractor`).

Exceptions in utility are meant to be ones used across multiple use cases, such as `ValidationException` that applies for every use case that requires database access. Exceptions used for a specific use case is defined within its interactor.